### PR TITLE
Created ShoppingCart and CartItem model. Added ShoppingCart controller

### PR DESCRIPTION
**Pull request by GlebPashko**

### CODE REVIEW

File: src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartResponseDto getShoppingCart() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        User user = userRepository.findByEmail(authentication.getName())
+                .orElseThrow(() -> new EntityNotFoundException("User not found"));
+
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId());
+        shoppingCart.setCartItems(cartItemRepository.findByShoppingCartId(shoppingCart.getId()));
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java:37]:
```suggestion
```

File: src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId());
+
+        CartItem cartItem = cartItemMapper.toModel(requestDto);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItemRepository.save(cartItem);
+    }
+
+    @Override
+    public void updateQuantity(Long id, UpdateShoppingCartRequestDto requestDto) {
+        CartItem cartItem = cartItemRepository.findById(id)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java:57]:
```suggestion
        ShoppingCart cart = repository.findByUserId(userId);
        CartItem cartItem = itemRepository.findByIdAndShoppingCartId(cartItemId, cart.getId())
```

update quantity

File: src/main/resources/db/changelog/changes/12-create-shopping_cart_cart-items-table.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: shopping_cart_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_shoppingcartcartitems_shopping_carts
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/12-create-shopping_cart_cart-items-table.yml:21]: bad naming for fk (not readable)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Override
+    public ShoppingCartResponseDto getShoppingCart() {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId());
+
+        return shoppingCartMapper.toShoppingCartResponseDto(shoppingCart);
+    }
+
+    @Override
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java:41]:
```suggestion
    @Override
    @Transactional
```

File: src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    public ShoppingCartResponseDto addBookToShoppingCart(CartItemRequestDto requestDto) {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Book book = bookRepository.findById(requestDto.getBookId())
+                .orElseThrow(() -> new NoSuchElementException("Book not found"));
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java:45]:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Book not found"));
```

provide book id to message

File: src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                    CartItem newCartItem = cartItemMapper.toModel(requestDto);
+                    newCartItem.setShoppingCart(shoppingCart);
+                    newCartItem.setBook(book);
+                    return newCartItem;
+                });
+        cartItem.setQuantity(cartItem.getQuantity() + requestDto.getQuantity());
+        cartItemRepository.save(cartItem);
+        shoppingCart.getCartItems().add(cartItem);
+
+        return shoppingCartMapper.toShoppingCartResponseDto(shoppingCart);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/ShoppingCartServiceImpl.java:62]: check in postman that you get shop cart with cartItems (and in DB also)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**